service/vsphere.go

```diff
@@ -90,17 +91,32 @@ func (vmh *VMHost) getMACAddressOfVM(vm *object.VirtualMachine) (string, error)
 	return vmDeviceList.PrimaryMacAddress(), nil
 }
 
-func getLocalMAC() (string, error) {
-	ifs, err := net.Interfaces()
+func getLocalMAC(ifaceListFunc func() ([]net.Interface, error), ifaceIncludeFilter *regexp.Regexp, ifaceExcludeFilter *regexp.Regexp) (string, error) {
+	ifs, err := ifaceListFunc()
 	if err != nil {
 		return "", err
 	}
 	for _, v := range ifs {
-		if v.HardwareAddr.String() != "" {
-			return v.HardwareAddr.String(), nil
+		if ifaceExcludeFilter != nil || ifaceIncludeFilter != nil {
+			if ifaceIncludeFilter != nil && ifaceIncludeFilter.MatchString(v.Name) {
```

Comment:
elaborate on the use-case for having the ifaceIncludeFilter? 
anyhow all VMs except excludefilter will be selected